rest/authorization-server/src/main/java/org/eclipse/sw360/rest/authserver/security/Sw360AuthenticationProvider.java

```diff
@@ -153,4 +176,32 @@ private boolean isDevEnvironment() {
     private boolean isValidString(String string) {
         return string != null && string.trim().length() != 0;
     }
+
+    public OAuth2AccessToken generateAccessTokenByUser(User user) {
+        Authentication authenticationToken = createAuthenticationToken(user.getEmail(), "...", user);
```

Comment:
Slightly offtopic:
Why is the password a required argument to generate the `authenticationToken`? This looks confusing and in https://github.com/eclipse/sw360/pull/352/files#diff-78934ad538171999c3e7ee21d8041918R163 (part of the  we actually use the real password in some way to generate the token??? We should clearify that.

Reply:
Yes `"..."` is the password.
Currently we use this token type `UsernamePasswordAuthenticationToken`.
We can introduce this in addition: `PreAuthenticatedAuthenticationToken`.
See my code corrections.

In the current implementation we need to have full client credentials (username + password)
to verify if the user and create a `UsernamePasswordAuthenticationToken`.

Reply:
> In the current implementation we need to have full client credentials (username + password)
to verify if the user and create a `UsernamePasswordAuthenticationToken`.

But the code does the authentication against liferay and then passes the password on to the next calls. This is unnecessary and should be removed, to prevent missuse of the password. As you show with using `"..."` the password is unused.
see: https://github.com/eclipse/sw360/blob/2e0334863c675bb4ee3ef2b38cf9ae0fc8155e0b/rest/authorization-server/src/main/java/org/eclipse/sw360/rest/authserver/security/Sw360AuthenticationProvider.java#L100-L105

My question is now: why does the `createAuthenticationToken` asks for the password, resp. if the constructor of `UsernamePasswordAuthenticationToken` gets the password, it might add the it to the token in some way.